File: campus-navigate-app/flutter_app/lib/domain/navigation/derived_navigation_estimate.dart
enum DerivedNavigationEstimateSource {
  debugReplay('debug-replay'),
  externalDerived('external-derived'),
  manualTest('manual-test'),
  pdrSummary('pdr-summary');

  const DerivedNavigationEstimateSource(this.wireValue);
  final String wireValue;
}

final class DerivedNavigationEstimate {
  const DerivedNavigationEstimate({
    required this.confidence,
    required this.headingDegrees,
    required this.source,
    required this.timestampMs,
    required this.x,
    required this.y,
  });

  final double confidence;
  final double headingDegrees;
  final DerivedNavigationEstimateSource source;
  final int timestampMs;
  final double x;
  final double y;
}

File: campus-navigate-app/expo-app/src/navigationInput/type/DerivedNavigationEstimate.ts
export type DerivedNavigationEstimate = {
  confidence: number;
  headingDegrees: number;
  source: 'debug-replay' | 'external-derived' | 'manual-test' | 'pdr-summary';
  timestampMs: number;
  x: number;
  y: number;
};
